azure/services/virtualmachineimages/images_test.go

```diff
@@ -0,0 +1,488 @@
+					{Name: to.StringPtr("121.13.20220524")},
+				},
+			},
+		},
+		{
+			k8sVersion:      "v1.22.0",
+			expectedSKU:     "k8s-1dot22dot0-ubuntu-2004",
+			expectedVersion: "latest",
+		},
+		{
+			k8sVersion:      "v1.22.9",
+			expectedSKU:     "k8s-1dot22dot9-ubuntu-2004",
+			expectedVersion: "latest",
+		},
+		{
+			k8sVersion:      "v1.22.10",
+			expectedSKU:     "ubuntu-2004-gen1",
+			expectedVersion: "122.10.20220613",
+			versions: compute.ListVirtualMachineImageResource{
+				Value: &[]compute.VirtualMachineImageResource{
+					{Name: to.StringPtr("122.10.20220524")},
+					{Name: to.StringPtr("122.10.20220613")},
+				},
+			},
+		},
+		{
+			k8sVersion:      "v1.23.6",
+			expectedSKU:     "k8s-1dot23dot6-ubuntu-2004",
+			expectedVersion: "latest",
+		},
+		{
+			k8sVersion:      "v1.23.7",
+			expectedSKU:     "ubuntu-2004-gen1",
+			expectedVersion: "123.7.20231231",
+			versions: compute.ListVirtualMachineImageResource{
+				Value: &[]compute.VirtualMachineImageResource{
+					{Name: to.StringPtr("123.7.20221124")},
+					{Name: to.StringPtr("123.7.20220524")},
+					{Name: to.StringPtr("123.7.20231231")},
+					{Name: to.StringPtr("123.7.20220818")},
+				},
+			},
+		},
+		{
+			k8sVersion:      "v1.24.0",
+			expectedSKU:     "ubuntu-2004-gen1",
+			expectedVersion: "124.0.20220512",
+			versions: compute.ListVirtualMachineImageResource{
+				Value: &[]compute.VirtualMachineImageResource{
+					{Name: to.StringPtr("124.0.20220512")},
+				},
+			},
+		},
+	}
+
+	location := "westus3"
+	for _, test := range tests {
+		test := test
+		t.Run(test.k8sVersion, func(t *testing.T) {
+			t.Parallel()
+
+			mockCtrl := gomock.NewController(t)
+			defer mockCtrl.Finish()
+			mockAuth := mock_azure.NewMockAuthorizer(mockCtrl)
+			mockAuth.EXPECT().HashKey().Return(t.Name()).AnyTimes()
+			mockAuth.EXPECT().Authorizer().AnyTimes()
+			mockAuth.EXPECT().BaseURI().AnyTimes()
+			mockAuth.EXPECT().SubscriptionID().AnyTimes()
+			mockClient := mock_virtualmachineimages.NewMockClient(mockCtrl)
+			svc := Service{Client: mockClient, Authorizer: mockAuth}
+
+			if test.versions.Value != nil {
+				mockClient.EXPECT().
+					List(gomock.Any(), location, azure.DefaultImagePublisherID, azure.DefaultImageOfferID, gomock.Any()).
+					Return(test.versions, nil)
+			}
+			image, err := svc.GetDefaultUbuntuImage(context.TODO(), location, test.k8sVersion)
+
+			g := NewWithT(t)
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(image.Marketplace.Version).To(Equal(test.expectedVersion))
+			g.Expect(image.Marketplace.SKU).To(Equal(test.expectedSKU))
+		})
+	}
+}
+
+func TestGetDefaultWindowsImage(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	mockClient := mock_virtualmachineimages.NewMockClient(mockCtrl)
+	svc := Service{Client: mockClient}
+
+	var tests = []struct {
+		name          string
+		k8sVersion    string
+		runtime       string
+		osVersion     string
+		expectedSKU   string
+		expectedError bool
```

Comment:
Can we change `expectedError` to a string containing the error message? That way we can verify that the correct error is being produced.